experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/node.ts

```diff
@@ -4,25 +4,66 @@
  */
 
 import { fail } from "../../../util";
+import { TreeNodeSchema } from "../../typed-schema";
+import { EditableTreeEvents } from "../../untypedTree";
 import { TreeNode } from "../editableTreeTypes";
-import { SharedTreeNode } from "./types";
-
-export const nodeSym = Symbol("node");
+import { getProxyForNode } from "./proxies";
+import { ProxyNode, SharedTreeNode, getTreeNode } from "./types";
 
 /**
- * Exposes node functionality, such as event subscriptions, for the given SharedTreeNode.
+ * The `node` object holds various functions for analyzing {@link SharedTreeNode}s.
  * @alpha
  */
-export function node(owner: SharedTreeNode): Required<SharedTreeNode>[typeof nodeSym] {
-	return (
-		(owner[nodeSym] as Required<SharedTreeNode>[typeof nodeSym]) ??
-		fail("owner must be a SharedTreeNode")
-	);
-}
+export const node = {
+	/**
+	 * The schema information for this node.
+	 */
+	schema: (n: SharedTreeNode) => {
+		return assertTreeNode(n).schema;
+	},
+	/**
+	 * Narrow the type of the given object if it satisfies the given schema.
+	 * @example
+	 * ```ts
+	 * if (node.is(myNode, point)) {
+	 *     const y = myNode.y; // `myNode` is now known to satisfy the `point` schema and therefore has a `y` coordinate.
+	 * }
+	 * ```
+	 */
+	is: <TSchema extends TreeNodeSchema>(u: unknown, schema: TSchema): u is ProxyNode<TSchema> => {
```

Comment:
I'm guessing you didn't want to use 'node' here since it's legal/useful to invoke this API on objects that are potentially not tree nodes?

I often see 'value' for any/unknown arguments and 'obj' if the input is expected to be an object.

Reply:
I renamed all the parameters to "node", and I renamed "node" to "nodeApi" - but only at in this module. The first index.ts aliases it back to "node".

Reply:
I meant to make the reply on your other comment about "node". Thanks for the suggestion here, I like `value` much better.